codec-haproxy/src/main/java/io/netty/contrib/handler/codec/haproxy/HAProxyMessage.java

```diff
@@ -270,21 +256,20 @@ private static HAProxyTLV readNextTLV(final ByteBuf header) {
         final int length = header.readUnsignedShort();
         switch (type) {
         case PP2_TYPE_SSL:
-            final ByteBuf rawContent = header.retainedSlice(header.readerIndex(), length);
-            final ByteBuf byteBuf = header.readSlice(length);
-            final byte client = byteBuf.readByte();
-            final int verify = byteBuf.readInt();
+            final Buffer rawContent = header.split(length);
```

Comment:
I think this should be `readSplit(length)`, since the length is from the current reader offset.

Reply:
Previously we have

```
final ByteBuf rawContent = header.retainedSlice(header.readerIndex(), length);
final ByteBuf byteBuf = header.readSlice(length);
```

Now instead if creating this temporary `byteBuf` we have

```
final Buffer rawContent = header.split(length);
```

and continue to work with `header` buffer

Reply:
Still doesn't look right to me. Since we've read `typeAsByte` and `length`, the `readerOffset()` of `header` is at least 3. The `split(length)` call captures the region from offset `0` to `length`.
For initial reader index of `0`, the previous `retainedSlice` call instead captured the region `3` to `3+length`.

Reply:
@chrisvest Please check the new commit